support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,243 @@
+        this.insertStStmt = session.prepare("insert into " + ST_TABLE + " (id, ticket) values (?, ?) ").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.deleteStStmt = session.prepare("delete from " + ST_TABLE + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.updateStStmt = session.prepare("update " + ST_TABLE + " set ticket = ? where id = ? ").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+        this.selectExStmt = session.prepare("select ticket, id from " + EXPIRY_TABLE + " where expiration_bucket = ? ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.selectDateExStmt = session.prepare("select expiration_bucket from " + EXPIRY_TABLE)
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        this.selectLrStmt = session.prepare("select last_run from " + LAST_RUN_TABLE + " where id = 'LASTRUN' ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.updateLrStmt = session.prepare("update " + LAST_RUN_TABLE + " set last_run = ? where id = 'LASTRUN' ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        final long lastRun = getLastRunTimestamp();
+        final long currentTime = currentTimeBucket();
+        if (lastRun == 0 || lastRun > currentTime) {
+            updateLastRunTimestamp(currentTime);
+        }
+    }
+
+    @Override
+    public void addTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("Inserting ticket {}", ticketId);
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        final String storageName = ticketDefinition.getProperties().getStorageName();
+
+        if (TGT_TABLE.equals(storageName)) {
+            final String parentTgtId = ticket.getGrantingTicket() == null ? null : ticket.getGrantingTicket().getId();
+            session.execute(this.insertTgtStmt.bind(ticket.getId(), serializer.serialize(ticket), parentTgtId, calculateExpirationDate(ticket) / TEN));
+        } else if (ST_TABLE.equals(storageName)) {
+            session.execute(this.insertStStmt.bind(ticket.getId(), serializer.serialize(ticket)));
+        } else {
+            LOGGER.error("Inserting unknown ticket type {}", ticket.getClass().getName());
+        }
+    }
+
+    @Override
+    public long deleteAll() {
+        return TICKETS_IN_TESTS;
+    }
+
+    @Override
+    public boolean deleteSingleTicket(final String ticketId) {
+        LOGGER.debug("Deleting ticket {}", ticketId);
+        if (ticketId == null) {
+            return false;
+        }
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        final String storageName = ticketDefinition.getProperties().getStorageName();
+        if (TGT_TABLE.equals(storageName)) {
```

Comment:
All conditional logic such as this needs to go away. You want to look up the relevant statement from a cache based on the storage name and/or ticket type rather than hardcoding checks. You are not just dealing with TGTs and STs and checking for types manually does not simply scale.